cluster/node.go

```diff
@@ -75,18 +117,150 @@ func (n *Node) SetState(s NodeState) {
 	n.Unlock()
 }
 
-// provide thread safe json Marshaling
-func (n *Node) MarshalJSON() ([]byte, error) {
+func (n *Node) SetPartitions(part []int32) {
```

Comment:
This could be called at any time, but it would only make sense for the input plugin to call it as that is what controls the partitions that are being ingested.   For example if the kafkamdm input plugin is configured to consume from "*" and we update the kafka topic to use more partitions it should call SetPartitions()